x-pack/plugins/security_solution/public/flyout/entity_details/host_right/hooks/use_observed_host.ts

```diff
@@ -15,11 +17,20 @@ import { Direction, NOT_EVENT_KIND_ASSET_FILTER } from '../../../../../common/se
 import { HOST_PANEL_OBSERVED_HOST_QUERY_ID, HOST_PANEL_RISK_SCORE_QUERY_ID } from '..';
 import { useQueryInspector } from '../../../../common/components/page/manage_query';
 import type { ObservedEntityData } from '../../shared/components/observed_entity/types';
+import { isActiveTimeline } from '../../../../helpers';
 
 export const useObservedHost = (
-  hostName: string
+  hostName: string,
+  scopeId: string
 ): Omit<ObservedEntityData<HostItem>, 'anomalies'> => {
-  const { to, from, isInitializing, setQuery, deleteQuery } = useGlobalTime();
+  const timelineTime = useDeepEqualSelector((state) =>
+    inputsSelectors.timelineTimeRangeSelector(state)
+  );
+  const globalTime = useGlobalTime();
+  const isActiveTimelines = isActiveTimeline(scopeId);
+  const { to, from } = isActiveTimelines ? timelineTime : globalTime;
+  const { isInitializing, setQuery, deleteQuery } = globalTime;
+
```

Comment:
This is probably not taking the correct sourcerer patterns either. Should we do:

`const { selectedPatterns } = useSourcererDataView(getSourcererScopeId(scopeId));`

Reply:
I mean, line 34

Reply:
I missed that. Logging the values, I see a small difference on the alerts page as well as from timeline, so I just want to highlight it:

values when the flyout is opened from the alerts page
![Screenshot 2024-05-08 at 11 08 55 AM](https://github.com/elastic/kibana/assets/17276605/9b83f0b2-71a6-4695-95c8-c70ed00a6bd0)

values when the flyout is opened from timeline
![Screenshot 2024-05-08 at 11 09 08 AM](https://github.com/elastic/kibana/assets/17276605/9ac38dba-60c7-49bd-b864-6459fac260e5)

Is that really what we want?

Reply:
Yes, these patterns depend on what the user has set in the data view, they look correct to me.

<img width="648" alt="timeline_patterns" src="https://github.com/elastic/kibana/assets/17747913/e426d5d3-aaba-4b03-a8c6-6a5b722b8046">

Reply:
yes of course, sorry! Added in [this commit](https://github.com/elastic/kibana/pull/182968/commits/c55a87e537d02fd60eaa4fdf0dd8f1266c1a72e1)